samples/java/src/main/java/com/gradle/enterprise/export/ExportApiJavaExample.java

```diff
@@ -1,119 +1,211 @@
 package com.gradle.enterprise.export;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.base.Joiner;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
-import io.netty.buffer.ByteBuf;
-import io.reactivex.netty.protocol.http.client.HttpClient;
-import io.reactivex.netty.protocol.http.client.HttpClientRequest;
-import io.reactivex.netty.protocol.http.client.HttpClientResponse;
-import io.reactivex.netty.protocol.http.sse.ServerSentEvent;
-import rx.Observable;
-import rx.exceptions.Exceptions;
+import com.google.common.util.concurrent.MoreExecutors;
+import okhttp3.ConnectionPool;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSourceListener;
+import okhttp3.sse.EventSources;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 import java.io.IOException;
-import java.net.InetSocketAddress;
-import java.net.SocketAddress;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.Base64;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
-import java.util.Set;
-import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
 
 import static java.time.Instant.now;
 
 public final class ExportApiJavaExample {
 
-    private static final SocketAddress GRADLE_ENTERPRISE_SERVER = new InetSocketAddress("gradle.my-company.com", 443);
-    private static final String EXPORT_API_USERNAME = "username";
-    private static final String EXPORT_API_PASSWORD = "password";
-    private static final String EXPORT_API_CREDENTIALS = Base64.getEncoder().encodeToString((EXPORT_API_USERNAME + ":" + EXPORT_API_PASSWORD).getBytes());
+    private static final HttpUrl GRADLE_ENTERPRISE_SERVER_URL = HttpUrl.parse("https://e.grdev.net");
```

Comment:
I don't think we should commit the e.grdev.net address, rather a truly public one, or just a fake one, to be replaced.

Reply:
Oops! This one slipped in from my own testing.

It's tempting to make this an env var, or to have it passed in as a CLI argument like it is with the Javascript example.